test/tests/functional/pbs_basil_support.py

```diff
@@ -362,5 +407,393 @@ def test_hbmemm_rsc(self):
+
+        # Check that the nodes are now stale.
+        self.server.expect(VNODE, {'state': 'Stale'}, id=k1, interval=5)
+        self.server.expect(VNODE, {'state': 'Stale'}, id=k2)
+
+        # Change nodes back to batch mode and restart PBS
+        cmd = ['xtprocadmin', '-k', 'm', 'batch']
+        self.logger.info(cmd)
+        ret = self.server.du.run_cmd(self.server.hostname,
+                                     cmd, logerr=True)
+        self.assertEqual(ret['rc'], 0)
+
+        # Restart Mom with signal HUP
+        self.mom.signal('-HUP')
+
+        # Check that the nodes are now stale.
+        self.server.expect(VNODE, {'state': 'free'}, id=k1, interval=5)
+        self.server.expect(VNODE, {'state': 'free'}, id=k2)
+
+        # Restore hook freq to 300
+        self.server.manager(MGR_CMD_SET, PBS_HOOK,
+                            {'enabled': 'true', 'freq': 300},
+                            id='PBS_alps_inventory_check', expect=True)
+
+    def test_job_run_on_knl_node(self):
+        """
+        Change the mode of KNL nodes to batch.
+        Then check if the PBS_alps_inventory_check hook picks up on the change.
+        Submit job and confirm job should be in R state
+        """
+        if self.du.platform == 'craysim':
+            self.skipTest(reason='xtprocadmin cmd is not on cray simulator')
+
+        # Find the list of KNL vnodes
+        klist = self.retklist()
+        if len(klist) == 0:
+            self.skipTest(reason='No KNL vnodes present')
+            return
+        else:
+            self.logger.info("KNL vnode list: %s" % (klist))
+
+        # Set pbshook frequency to 10 seconds
+        self.server.manager(MGR_CMD_SET, PBS_HOOK,
+                            {'enabled': 'true', 'freq': 10},
+                            id='PBS_alps_inventory_check', expect=True)
+
+        # Restart Mom with signal HUP
+        self.mom.signal('-HUP')
+
+        now = time.time()
+
+        # Change mode of all nodes to interactive
+        cmd = ['xtprocadmin', '-k', 'm', 'interactive']
+        self.logger.info(cmd)
+        ret = self.server.du.run_cmd(self.server.hostname,
+                                     cmd, logerr=True)
+        self.assertEqual(ret['rc'], 0)
+
+        # Change mode of two KNL nodes to batch
+        if len(klist) >= 2:
+            k1 = klist[0]
+            k2 = klist[len(klist) - 1]
+            knl1 = int(re.search(r'\d+', k1).group())
+            knl2 = int(re.search(r'\d+', k2).group())
+
+        cmd = ['xtprocadmin', '-k', 'm', 'batch', '-n', "%s" % (knl1)]
+        self.logger.info(cmd)
+        ret = self.server.du.run_cmd(self.server.hostname,
+                                     cmd, logerr=True)
+        self.assertEqual(ret['rc'], 0)
+
+        cmd = ['xtprocadmin', '-k', 'm', 'batch', '-n', "%s" % (knl2)]
+        self.logger.info(cmd)
+        ret = self.server.du.run_cmd(self.server.hostname,
+                                     cmd, logerr=True)
```

Comment:
I think you can do these two commands in one by passing -n a comma separated list.